async/retry.ts

```diff
@@ -119,15 +119,15 @@ export interface RetryOptions {
  */
 export async function retry<T>(
   fn: (() => Promise<T>) | (() => T),
-  options?: RetryOptions,
+  options: RetryOptions = {},
```

Comment:
Why do we need this change?

BTW there's an occurrence of the same structure in `unstable_retry.ts`

Reply:
For some reason the options setup lines were not marked as covered in `codecov` in a previous PR before this change. Might have been a bug though.

Reply:
`retry` seems covered 100% in codecov including option setup line https://app.codecov.io/gh/denoland/std/blob/main/async%2Fretry.ts